docs/dynamicproxy-default-parameter-values.md

```diff
@@ -0,0 +1,43 @@
+# Some default parameter values may not be reproduced correctly in proxy types
+
+DynamicProxy uses `System.Reflection` and `System.Reflection.Emit` to create proxy types. Due to several bugs and limitations in these facilities (both on .NET and Mono), it is not possible in every single case to faithfully reproduce default values of optional parameters in the dynamically generated proxy types.
+
+This usually doesn't matter much in practice, but it may become a problem for frameworks or libraries (such as Dependency Injection containers) that reflect over the generated proxy types.
+
+
+## When your code runs on Mono
+
+On Mono (up to and including at least version 5.10.1.47), DynamicProxy may not be able to correctly reproduce default parameter values in the proxy type for...
+
+* **Optional parameters of any nullable type `Nullable<T>`.** Reflection will likely report (via `ParameterInfo.[Raw]DefaultValue`) a default value of `null` for such parameters, regardless of whether the actual default value in the proxied method was `null` or not.
+
+   Therefore, you cannot trust the reported default value, *unless* you know you're running on a version of Mono where the underlying issues have been resolved, or when you've double-checked the default value in the original method of the proxied type.
+
+   The underlying causes have been documented in [mono/mono#8504](https://github.com/mono/mono/issues/8504) and [mono/mono#8597](https://github.com/mono/mono/issues/8597).
+
+
+## When your code runs on .NET or .NET Core
```

Comment:
".NET" should be ".NET Framework" (there are a few mentions below), it is important to identity the product especially when you then refer to ".NET Core" in the same sentence.